test/Microsoft.AspNet.Mvc.ViewFeatures.Test/Rendering/HtmlHelperValidationMessageExtensionsTest.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.AspNet.Mvc.TestCommon;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.Core
+{
+    /// <summary>
+    /// Test the ValidationMessage extensions in <see cref="HtmlHelperValidationExtensions" /> class.
+    /// </summary>
+    public class HtmlHelperValidationMessageExtensionsTest
```

Comment:
For another PR: This test class covers all extension methods but only when the property is valid and client-side validation is enabled. Need to confirm what happens for all combinations of those two characteristics e.g. invalid properties cause a `<span>` to be generated even if client-side validation is disabled.